Ratwood Mage Gameplay Loop Port (Ready to TM :tm:)

WeNeedMorePhoron opened the pull request:
## About The Pull Request
This PR is basically a port of the Ratwood Mage Gameplay Loop from https://github.com/Rotwood-Vale/Ratwood-Keep/pull/2213

All credits to RadiantFlash07 for the original PR and I am happy to finally see this content getting here.

Contents includes: 
- Obtaining manacrystals, manabloom flowers, obsidian shards, runed artifact,s leyline shards
- Every summons is now summonable after we ported it from Vanderlin who ported it from RW. Probably bindable (didn't test)
- Enchantment are ported from Ratwood as is, with the following skipped / removed: Woundclosing, Phoenix guard, temporal rewind, freezing, infernal flame, leaping.
- Recipe Books integration 
- Giving the new chalk to Witch (removing ritualist), Artificer, Mage Apprentice, Hedge Mace. Court Magos now get a silver dagger for instant access to up to T4 ritual.
- All of the Arcane Crafting including binding shackles and inferna lforge etc. Except the amulet that give you arcana.
- Wall Accession & Fortress Accession Matrix
- Yoinked some icons from Ratwood to fill in for our spells, including Mending, Darkvision, Enlarge, nondetection
- Ported Game Satchel from Vanderlin. Hunter lose their Messer, gain a gamebag to hold meat.

**Chalk / Silver Dagger:**
- Chalk / Silver Dagger is acquired by Arcana Crafting / Applying a Cinnabar to a Silver Dagger respecitvely (more on this later)

**Mapping Changes:**
- Guaranteed Mana Fountain in the western bog
- Guaranteed Leyline in the eastern bog 
- Swapped out runes in Orc and Goblin Fortress Dungeon for functional one 
- **FieryTech**  - Remapped Mage Tower basement and Guild Basement for a Summoning + Imbuement Array

**Contents not included:**
- The spellbook grinding + spellpoint gain loop (Probably, i might've missed something)
- Rune Rituals for buffing stats and planar convergence matrix (teleport) 
- Any spells 
- Any of the aesthetics thing


<!-- Describe your pull request. Avoid text walls, use concise bullet points for easier readability. Document every change, or this can delay review and even discourage maintainers from merging your PR. -->

## Testing Evidence
![dreamseeker_DxrZK1gK72](https://github.com/user-attachments/assets/8c30830b-4c10-4854-98a5-2aa169ce269e)
![StrongDMM_hCahzDZxWv](https://github.com/user-attachments/assets/85051c91-2886-4d5a-b91e-8758e9634dcc)
![StrongDMM_yyknYYlEro](https://github.com/user-attachments/assets/a3396788-d432-40f6-9a1c-eb5b90b9d674)


<!-- It's mandatory to test your PR. Provide images, clips or description of how you tested your changes where possible. -->

## Why It's Good For The Game
Shit for Mages to do, finally alleviating my burnout, perhaps? 

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial. If you can't, then it probably isn't good for the game in the first place. -->

WeNeedMorePhoron committed:
Stash Local

WeNeedMorePhoron committed:
Compile

WeNeedMorePhoron committed:
Compile - 2

WeNeedMorePhoron committed:
Add Icons. Probably.

WeNeedMorePhoron committed:
Gas

WeNeedMorePhoron committed:
gas

WeNeedMorePhoron committed:
Oops compile

WeNeedMorePhoron committed:
Uhh makes it work I guess.

AllCorruptingWyrm commented:
awesome

Eagletanker commented:
Will witches still be able to do rituals calling upon their patrons or are they just mage without combat spells (and thus can't use the ritual system)

PrettyPsychoForAWolf commented:
Will the chalk/dagger be craftable like in RW? Cause otherwise this'll be fun. 

Though I'm preeetty sure this is going to be hit with some major nerfs over time. A lot of the stuff in that gameplay loop is broken good. (Not referring to the things you already removed btw)

WeNeedMorePhoron commented:
There'll be using the new Arcane ritual system

NalimaTheWitch commented:
sire, it's kino

WeNeedMorePhoron committed:
Remap path. Yippee.

WeNeedMorePhoron commented:
I'll fix up the linters error later (another day)

chemistrymain3 commented:
king shit

WeNeedMorePhoron committed:
Fix linter issue. TM

WeNeedMorePhoron committed:
Some misc. fix, remove leaping.

radarseas commented:
I'd add an expensive way for mages to get new chalks, because I can DEFINITELY foresee people losing these to being confiscated without return for summoning dangerous things, but mages should still have a way to replace them.

WeNeedMorePhoron commented:
@radarseas Chalk are already craftable under Arcana Crafting - the NEW chalk, that is. 

In the future there might need to be balancing w/ Mages sacrificing spell points to acquire summon but it can be done later.

WeNeedMorePhoron commented:
The silver dagger can be made with a silver dagger and cinnabar

WeNeedMorePhoron committed:
FieryTech map changes

WeNeedMorePhoron committed:
Remove Towner Messer for Meatbag.

WeNeedMorePhoron committed:
Merge branch 'main' of https://github.com/GeneralPantsuIsBadAtCoding/Azure-Peak into mage-gameplay-loop-port-2

DorfGuy1 commented:
The mining enchantment has the potential to blow our economy up even more.

WeNeedMorePhoron commented:
I don't think this will be TMed in the next day but I appreciate the headsup. I'll take a look at how strong the enchantments are. Although here Masterwork / Item quality are mostly aesthetics or for sell value.

I'll check the T4 Fire Boss again. We mapped in up to T3 and used the whole sets. Only the void dragon have a version of the meteor storm atm

PrettyPsychoForAWolf commented:
> Although here Masterwork / Item quality are mostly aesthetics or for sell value.

I havent checked ratwood code in a hot minute (sadly cant for a few days either), but I dont think they use the quality from smithing to judge hammer quality. They use different quality hammers. (Like stone, steel, etc) Don't quote me on that cause my memory's fuzzy. 

Essentially that code in the pr I linked is a groundwork for other hammer types.

WeNeedMorePhoron committed:
Deconflict

WeNeedMorePhoron committed:
Add summoner pouch sprite jian huo.

WeNeedMorePhoron committed:
Fix recipes

WeNeedMorePhoron committed:
Manabloom sprite

WeNeedMorePhoron committed:
Format the basic section better.

WeNeedMorePhoron committed:
git push
Merge branch 'mage-gameplay-loop-port-2' of https://github.com/WeNeedMorePhoron/Azure-Peak into mage-gameplay-loop-port-2

SovietSpaghetti commented:
This is great so far! Only thing I can really think of QoL-wise is being able to pick up everything on a tile using the summoner pouch like a sack, and maybe expanding the item spawns outside of just the swamp (which is kinda hell for casters here). If it's a difficulty-gating thing, with how big mount decap is here, they could easily spawn there.

WeNeedMorePhoron committed:
Add Cinnabar to stockpile.

WeNeedMorePhoron committed:
Deslapcraftify the shackles.

WeNeedMorePhoron committed:
Fix Obsidian and Leyline shard being flammable. Add back in hellfather.

WeNeedMorePhoron committed:
Prevent mimic trinket from being reverted w/o changes

WeNeedMorePhoron committed:
Merge branch 'main' of https://github.com/GeneralPantsuIsBadAtCoding/Azure-Peak into mage-gameplay-loop-port-2

WeNeedMorePhoron committed:
Make runed artifact craftable.

WeNeedMorePhoron committed:
Cinnabar to stockpile, 2x limit. Make runed artifact craftable. Quickdumpable and storeable summoners pouch.

WeNeedMorePhoron committed:
Wild plant / mana bloom to make it slightly easier. Map edit for alternative sources of manabloom and two guarantee in guild.

WeNeedMorePhoron commented:
QOL added for the summoner pouch for quick gather and quick dumping.

Added some more guaranteed mana bloom spawn around the map - shitty map edit - one more alternative mana fountain in Mount Decap where the fox cave is. 

Mana Flower is a normal plant and not herb.

WeNeedMorePhoron committed:
Wild plants naming.

Varoxus commented on _maps/map_files/templates/runeritual_fortress/runeritual_fortress.dm:
This should be a DMM instead of a DM file... not even sure how this got put in here as a DM.

WeNeedMorePhoron committed:
Fix Arcyne Fortress and Infernal Forge.

WeNeedMorePhoron committed:
Remove Rune Fortress .dmm from .dme fuck

WeNeedMorePhoron committed:
Fix wild plants.

Ashturian commented:
> Witch'll be using the new Arcane ritual system. Chalk is craftable. Fairly sure that the dagger is craftable. But yeah, it will probably get nerfed and there will be unintended consequences and I will need to fix up the code etc. over time.
> 
> But we can bring it in first and adjust it later.

Witches losing ritualist for this is not really in exchange for much benefit - they can maybe enable the use of materials but they have very little ability to actually go out and gather them compared to other roles, where the ritualist runes at least have some immediate and minor boons

Any chance they could get ritualist back?
